application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/TripPatternForDate.java

```diff
@@ -224,23 +223,21 @@ public TripPatternForDate newWithFilteredTripTimes(Predicate<TripTimes> filter)
     return new TripPatternForDate(tripPattern, filteredTripTimes, filteredFrequencies, serviceDate);
   }
 
-  private static void assertValidRunningPeriod(
-    LocalDate startOfRunningPeriod,
-    LocalDate endOfRunningPeriod,
-    TripTimes first,
-    TripTimes last
-  ) {
-    if (first.getTrip().getRoute().getFlexibleLineType() != null) {
-      // do not validate running period for flexible trips
-      return;
-    }
+  /**
+   * Asserts that the running period is valid and throws an {@link IllegalArgumentException} if it
+   * is not.
+   * This is the last line of defense against invalid real-time updates.
```

Comment:
```suggestion
   * Asserts that the running period is valid and throws an {@link IllegalArgumentException} if it
   * is not. This is the last line of defense against invalid real-time updates.
```
We need to describe here why this is called from the RealTimeRaptorTransitDataUpdater and not encapsulated. So, something like:

> This validation is only needed for real-time updates and should not be applied to Flex trips ...

Reply:
But maybe we should encapsulate it. Perhaps a method like tripPatternForDateMapper#mapWithValidation which would run the validation. Then throwing the IllegalArgumentException feels a bit more natural?

Reply:
Can you discuss it in the dev meeting (I won't be there) and let me know?

Reply:
We discussed this in the dev meeting, you can refactor it as I suggested (we preferred that approach) but it's not a strict requirement.